draft-ietf-anima-brski-prm.md

```diff
@@ -2174,122 +2410,141 @@ When using a JWS signature, the Status Query artifact looks as shown in {{stat_r
   ]
 }
 ~~~~
-{: #stat_req title="Status Query Representation in General JWS JSON Serialization Syntax" artwork-align="left"}
-
-For details on `JWS Protected Header` and `JWS Signature` see {{!I-D.ietf-anima-jws-voucher}} or {{!RFC7515}}.
-
-
-### Response Artifact: Pledge Status (pStatus)
-
-When the pledge receives a Status Query with status-type `bootstrap` it SHALL respond with previously collected telemetry information (see {{vstatus}} and {{estatus}}) in a single Pledge Status artifact.
+{: #pstatus_representation title='pStatus Representation in General JWS JSON Serialization Syntax' artwork-align="left"}
 
+The JSON Pledge Status Data MUST be UTF-8 encoded to become the octet-based JWS Payload defined in {{RFC7515}}.
+The JWS Payload is further base64url-encoded to become the string value of the `payload` member as described in {{Section 3.2 of RFC7515}}.
+The octets of the UTF-8 representation of the JWS Protected Header are base64url-encoded to become the string value of the `protected` member.
+The generated JWS Signature is base64url-encoded to become the string value of the `signature` member.
 
-The pledge-status response message is signed with IDevID or LDevID, depending on bootstrapping state of the pledge.
+#### JSON Pledge Status Data {#pstatus_data}
 
-The following CDDL defines the structure of the Pledge Status (pStatus) data:
+The JSON Pledge Status Data SHALL be a JSON document {{RFC8259}} that MUST that MUST conform with the CDDL {{!RFC8610}} data model defined in {{stat_res_def}}:
 
 ~~~~ cddl
   pledgestatus = {
     "version": uint,
-    "status":
-      "factory-default" /
+    "status": $status-enumerations,
```

Comment:
The static CDDL tied to only the boostrap status-type felt improvised without extensions in mind.
I am not fully sure if it is correct CDDL now (I asked for help in a Telegram group with the CDDL authors...).
Further, I am not sure if I understood the intention around "status" correctly. To me, it seemed the allowed values for this member should depend on the status-type. So I expect an enum of allowed values for each status-type.

Reply:
Fixed CDDL.